astropy/time/formats.py

```diff
@@ -747,7 +759,7 @@ class TimeString(TimeUnique):
 
     def _check_val_type(self, val1, val2):
         # Note: don't care about val2 for these classes
-        if val1.dtype.kind not in ('S', 'U'):
+        if val1.dtype.kind not in ('S', 'U', 'f'):
```

Comment:
I see why you are doing this, but it is ugly! I think this does need to special-case the `shape=(0,)` case.

Reply:
More specifically, I think this would need to be
```
if val1.dtype.kind not in ('S', 'U') and val1.size > 0:
    raise ...
```